src/Google/Cache/File.php

```diff
@@ -151,11 +151,14 @@ private function getCacheDir($file, $forWrite)
     // use the first 2 characters of the hash as a directory prefix
     // this should prevent slowdowns due to huge directory listings
     // and thus give some basic amount of scalability
-    $dirHash = substr(md5($file), 0, 2);
+    $fileHash = substr(md5($file), 0, 2);
+    $userHash = md5(get_current_user());
+    $dirHash = $fileHash . DIRECTORY_SEPARATOR . $userHash;
```

Comment:
I don't want usernames showing in the directory path, it seems sloppy. Using the full `md5` brings the likelyhood of collisions to essentially zero.

I made the root directory `0777` and the individual ones `0700`. If this isn't acceptable, I'm going to remove this file all together and let everyone write their own implementations (kidding but kinda serious...)